go/vt/sqlparser/token.go

```diff
@@ -1059,7 +1063,20 @@ func (tkn *Tokenizer) scanMySQLSpecificComment() (int, []byte) {
 		if tkn.lastChar == eofChar {
 			return LEX_ERROR, buffer.Bytes()
 		}
+
 		tkn.consumeNext(buffer)
+
+		// Already found first letter since start of comment
+		if seenLetter {
+			continue
+		}
+
+		// Found first letter?
+		if isLetter(tkn.lastChar) {
```

Comment:
Getting closer, but you need to incorporate some of the logic in ExtractMysqlComment. The first non-directive character of the comment is the first non-space character following the 5-digit mysql version string.

It's not impossible that the first non-comment character is a digit or a quote character, for example